src/Avalonia.Controls/TopLevel.cs

```diff
@@ -824,12 +824,16 @@ private void HandleInput(RawInputEventArgs e)
         {
             if (PlatformImpl != null)
             {
-                if (e is RawPointerEventArgs pointerArgs)
+                Dispatcher.UIThread.Send(static state =>
                 {
-                    pointerArgs.InputHitTestResult = this.InputHitTest(pointerArgs.Position);
-                }
+                    var (topLevel, e) = (ValueTuple<TopLevel, RawInputEventArgs>)state!;
+                    if (e is RawPointerEventArgs pointerArgs)
+                    {
+                        pointerArgs.InputHitTestResult = topLevel.InputHitTest(pointerArgs.Position);
+                    }
 
-                _inputManager?.ProcessInput(e);
+                    topLevel._inputManager?.ProcessInput(e);
+                }, (this, e), DispatcherPriority.Send);
```

Comment:
I'm not sure if we should be changing the current DispatcherPriority to `Send`, it will affect the SynchronizationContext used for the callback.

Reply:
That's what WPF does.
Also, with other priorities dispatcher operation won't be executed immediately.

Reply:
See https://github.com/dotnet/wpf/blob/c799a93026bc5f6429258841f02a34fc2f6a8389/src/Microsoft.DotNet.Wpf/src/Shared/MS/Win32/HwndSubclass.cs#L338-L341

Reply:
Yes, but it will be a behavior change and make 
```
await Foo();
window.Close();
```
to cause native crashes on macOS in certain code paths (render being forced via `[NSView updateLayer]`) again. 

It was one of the reasons why we've changed our default dispatcher priority to be lower than Render.

Not sure how to proceed here, it's technically a behavior change.

Reply:
@kekekeks changed the way how `Dispatcher.Send` works. Also fixing behavioral change my previous changes introduced.
Both `AvaloniaSynchronizationContext.Send` and `Dispatcher.Send` shouldn't change current sync context, when we already on the dispatcher thread.